### PR TITLE
Promote CSP `frame-ancestors` directive

### PR DESCRIPTION
**Pull request by robhudson**
## One-line summary

The `frame-ancestors: 'none'` directive prevents the site from being embedded in any `<iframe>`, `<object>`, or `<embed>` elements, mitigating clickjacking attacks.

This directive has been running in report-only mode, generating reports without enforcing the policy. The reports show no valid embedding attempts, confirming that the directive will protect the website without breaking any legitimate use cases.

## Issue / Bugzilla link

#11943

### CODE REVIEW

File: bedrock/settings/__init__.py [hunk 1]
```diff
@@ -125,7 +126,6 @@
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["default-src"] = [csp.constants.SELF]
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["media-src"] = [csp.constants.SELF, "assets.mozilla.net", "videos.cdn.mozilla.net"]
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["object-src"] = [csp.constants.NONE]
-    CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["frame-ancestors"] = [csp.constants.NONE]
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["style-src"].remove(csp.constants.UNSAFE_INLINE)
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["upgrade-insecure-requests"] = True
```
Comment (janbrasna) [inline bedrock/settings/__init__.py:130]: I guess you can also enforce `upgrade-insecure-requests` as that otherwise has no effect in report-only mode.

Comment (robhudson) [reply bedrock/settings/__init__.py:130]: That was going to be my next PR. :) Trying to keep it simple and one directive at a time.

File: bedrock/settings/__init__.py [hunk 2]
```diff
@@ -156,20 +156,22 @@ def _override_csp(csp, append: dict[str, list[str]] = None, replace: dict[str, l
...
 if csp_ro_report_uri:
     # Path based overrides for report-only CSP.
-    CMS_ADMIN_CSP_RO = _override_csp(CONTENT_SECURITY_POLICY_REPORT_ONLY, replace={"frame-ancestors": [csp.constants.SELF]})
     CMS_ADMIN_IMAGES_CSP_RO = _override_csp(CONTENT_SECURITY_POLICY_REPORT_ONLY, append={"img-src": ["blob:"]})
 
     CSP_PATH_OVERRIDES_REPORT_ONLY = {
         # Order them from most specific to least.
         "/cms-admin/images/": CMS_ADMIN_IMAGES_CSP_RO,
-        "/cms-admin/": CMS_ADMIN_CSP_RO,
     }
```
Comment (janbrasna) [inline bedrock/settings/__init__.py:175]: Q out of ignorance: the admin images seem to be duplicated in both overrides — is that necessary due to the way the overrides are deepcopied and in which order, as no matter what is in enforced path overrides, that doesn't automagically translate into the equivalent RO definition at the moment?

Comment (robhudson) [reply bedrock/settings/__init__.py:175]: Yeah, we set the enforced CSP. Then deepcopy it to create the report-only CSP. Then the overrides use each of those as a base.

As a tree this looks like:
- CSP
  - CSP_RO
    - CSP_RO_OVERRIDES
  - CSP_OVERRIDES

Comment (janbrasna) [reply bedrock/settings/__init__.py:175]: So basically the path override rules should not be moved, but duplicated — not to disappear from the RO policy, sending false violations for the admin paths?

(Or, instead, the CSP_RO_OVERRIDES should start with a copy from CSP_OVERRIDES for consistency/predictability?)

Comment (janbrasna) [reply bedrock/settings/__init__.py:175]: IIUC if the current model stays, the removed CMS_ADMIN_CSP_RO here above need to be kept, to mirror what's being added to CMS_ADMIN_CSP, not to start triggering RO violations.

**Review by stevejalim — APPROVED** [review]

File: bedrock/settings/__init__.py (hunk 2 above)
Comment (robhudson) [reply bedrock/settings/__init__.py]: Yep, nice catch, thank you. That's fixed now. I also tried to move code together so it's a little easier to see what's going on.

**Pull request merged**